packages/adp-tooling/src/common/app-type.ts

```diff
@@ -0,0 +1,34 @@
+import type { Manifest } from '@sap-ux/project-access';
+import { ApplicationType } from '../types';
+
+/**
+ * Determines the type of UI5 application based on the content of its manifest file.
+ * This function checks various properties within the manifest to classify the application
+ * into predefined types such as Fiori Elements, Fiori Elements OVP, Free Style, or None.
+ *
+ * @param {Manifest} manifest - The manifest configuration object of the application.
+ * @returns {ApplicationType} The type of the application as defined by the ApplicationType enum.
+ *
+ * The classification is done based on the presence and values of specific keys in the manifest:
+ * - 'sap.ovp' indicates a Fiori Elements Overview Page (OVP).
+ * - 'sap.ui.generic.app' or 'sap.app' with a specific sourceTemplate id indicates a Fiori Elements application.
+ * - If none of these conditions are met, the function defaults to categorizing the application as Free Style.
+ * - If the manifest is empty, it returns None.
+ */
+export function getApplicationType(manifest: Manifest): ApplicationType {
```

Comment:
Why do we actually need the application type? I thought that was something we removed from the templates.

Reply:
We check the type because we need to see if the Adaptation Project supports it. These are supported:
FIORI_ELEMENTS = 'FioriElements',
FIORI_ELEMENTS_OVP = 'FioriElementsOVP',
FREE_STYLE = 'FreeStyle'
If it is something else we return an error stating that: `... Adaptation project doesn't support the selected application.`

Reply:
You return one of these three if `Object.keys(manifest).length > 0`, so technically, you validations just checks if the manifest is neither `undefined` nor `{}`.

Reply:
We removed this check for a valid application type here https://github.com/SAP/open-ux-tools/pull/2169/commits/3bb3bc6a249590729b79ba69812d876dd8bcb47d#diff-034445208bf7ca40b734423c5cf343d6d0a4761a4b55cee844776b5458a1c9a0L86-L102.

But since this is used in the writing part we did not remove the method. It is used in a method that builds the descriptor content: `createDescriptorChangeForResourceModel`.